pkg/controller/nodes/subworkflow/launchplan/admin.go

```diff
@@ -97,6 +97,26 @@ func (a *adminLaunchPlanExecutor) GetStatus(ctx context.Context, executionID *co
 	return item.ExecutionClosure, item.SyncError
 }
 
+func (a *adminLaunchPlanExecutor) GetLaunchPlan(ctx context.Context, launchPlanRef *core.Identifier) (*admin.LaunchPlan, error) {
```

Comment:
Having this method here is fine, but having a separate interface for this method is useful to differentiate the use cases and usage paths